packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,354 @@
+  varB: number,
+  meanB: number,
+  nB: number,
+  relative: boolean
+): number {
+  if (relative) {
+    return (
+      varB / (Math.pow(meanA, 2) * nB) +
+      (varA * Math.pow(meanB, 2)) / (Math.pow(meanA, 4) * nA)
+    );
+  } else {
+    return varB / nB + varA / nA;
+  }
+}
+
+export function powerStandardError(
+  variance: number,
+  mean: number,
+  n: number,
+  nVariations: number,
+  effectSize: number
+): number {
+  return Math.sqrt(
+    frequentistVariance(
+      variance,
+      mean,
+      n / nVariations,
+      variance,
+      mean * (1 + effectSize),
+      n / nVariations,
+      true
+    )
+  );
+}
+
+export function calculateRho(
+  alpha: number,
+  sequentialTuningParameter: number
+): number {
+  return Math.sqrt(
+    (-2 * Math.log(alpha) + Math.log(-2 * Math.log(alpha) + 1)) /
+      sequentialTuningParameter
+  );
+}
+
+export function sequentialPowerSequentialVariance(
+  variance: number,
+  n: number,
+  alpha: number,
+  sequentialTuningParameter: number
+): number {
+  const standardErrorSampleMean = Math.sqrt(variance / n);
+  const rho = calculateRho(alpha, sequentialTuningParameter);
+  const partUnderRadical =
+    (2 *
+      (n * Math.pow(rho, 2) + 1) *
+      Math.log(Math.sqrt(n * Math.pow(rho, 2) + 1) / alpha)) /
+    Math.pow(n * rho, 2);
+  const zSequential = Math.sqrt(n) * Math.sqrt(partUnderRadical);
+  const zStar = normal.quantile(1.0 - 0.5 * alpha, 0, 1);
+  const standardErrorSequential =
+    (standardErrorSampleMean * zSequential) / zStar;
+  const vSequential = Math.pow(standardErrorSequential, 2);
+  return n * vSequential;
+}
+
+export function sequentialPowerStandardError(
+  variance: number,
+  mean: number,
+  n: number,
+  nVariations: number,
+  effectSize: number,
+  alpha: number,
+  sequentialTuningParameter: number
+): number {
+  const v = sequentialPowerSequentialVariance(
+    variance,
+    n / nVariations,
+    alpha,
+    sequentialTuningParameter
+  );
+  return Math.sqrt(
+    frequentistVariance(
+      v,
+      mean,
+      n / nVariations,
+      v,
+      mean * (1.0 + effectSize),
+      n / nVariations,
+      true
+    )
+  );
+}
+
+/**
+ * Performs power calculation
+ *
+ * @param effectSize Scalar lift (relative to the scalar mean of the distribution, expressed as percentage).
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @param twoTailed Binary indicator if the test is 1 or 2-tailed (default: true).
+ * @returns Estimated power.
+ */
+export function powerEst(
+  effectSize: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  twoTailed: boolean = true,
+  sequentialTuningParameter = 0
+): number {
+  if (typeof twoTailed !== "boolean") {
+    throw new Error("twoTailed must be boolean.");
+  }
```

Comment:
note: This kind of check isn't really necessary since we rely on Typescript to ensure that whenever we pass something to this method that it is a boolean itself.

Leaving it in shouldn't be problematic at all though, and could provide a more informative error in the future; it's just very unlikely to ever trigger.

Reply:
removed this check